File: route_test.py
from Route import AUV,GRID,visualize,obstacles_inflation
import random
import time

def main():
    grid_size_x = 1000
    grid_size_y = 500
    pillars=[]
    for i in range(3):
        pillars.append((random.randint(1,999),random.randint(1,499)))

    obstacles = []
    for i in range(10):
        obstacles.append((random.randint(1,999),random.randint(1,499)))
    AUV_size = 30
    VELT=AUV((1,1), AUV_size)
    grid = GRID(grid_size_x, grid_size_y, pillars, obstacles)
    
    obstacles_inflation(grid, obstacles, AUV_size)
    
    start_time = time.perf_counter()    
    path = VELT.build_full_route(grid.targets, grid)
    end_time = time.perf_counter()
    print(end_time - start_time)
    if path:
        print("Path length:", len(path))
        visualize(grid.field, path, grid.targets)
main()

File: cpp/route_test.cpp
#include "Route.h"
#include <iostream>
#include <vector>
#include <chrono>
#include <random>
const double K_units = 20.0;

int main() {

    std::random_device rd; 
    std::mt19937 gen(rd());
    int min_x = 1, max_x = 999;
    std::uniform_int_distribution<> distr_x(min_x, max_x);
    int min_y = 1, max_y = 499;
    std::uniform_int_distribution<> distr_y(min_y, max_y);


    std::vector<Point> targets;
    targets.push_back({999, 499});
    targets.push_back({500, 250});
    targets.push_back({20, 400});

    double AUV_length = 1.0;
    double AUV_width = 0.4;
    double pool_length = 50.0;
    double pool_width = 25.0;
    double max_vel = 1.0;
    double min_vel = 0.1;
    
    Point start_pos = {1, 1};
    
    std::vector<Point> obstacles;
    for (int i = 0; i < 40; i++){
        int x = distr_x(gen);
        int y = distr_y(gen);
        Point point = {x, y};
        obstacles.push_back(point);
    }
    
    GRID grid(metres_to_grid_units(pool_length, K_units), metres_to_grid_units(pool_width, K_units), targets, obstacles);
    AUV VELT(start_pos, metres_to_grid_units(AUV_length, K_units), metres_to_grid_units(AUV_width, K_units), metres_to_grid_units(max_vel, K_units), metres_to_grid_units(min_vel, K_units));
    
    for (int i = 0; i < 40; i++){
        obstacles_inflation(grid.field, grid, obstacles[i], VELT.radius);
    }
    

    auto start = std::chrono::high_resolution_clock::now();
    std::vector<Point> full_route = VELT.build_full_route(targets, grid);
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;
    std::cout << duration.count() << std::endl;

    
    if (full_route.empty()) {
        std::cout << "Path Not Found" << std::endl;
    } else {
        std::cout << "X  Y  Angle  Velocity" << std::endl;
        angle_velocity_output(full_route, VELT.max_velocity, VELT.min_velocity);
    }


    return 0;
}